crates/extensions_ui/src/extension_api.rs

```diff
@@ -0,0 +1,153 @@
+use std::{path::PathBuf, sync::Arc};
+
+use anyhow::Result;
+use assistant_slash_command::SlashCommandRegistry;
+use extension::{extension_lsp_adapter::ExtensionLspAdapter, wasm_host};
+use fs::Fs;
+use gpui::{AppContext, BackgroundExecutor, Task};
+use indexed_docs::{IndexedDocsRegistry, ProviderId};
+use language::{LanguageRegistry, LanguageServerBinaryStatus, LoadedLanguage};
+use snippet_provider::SnippetRegistry;
+use theme::{ThemeRegistry, ThemeSettings};
+use ui::SharedString;
+
+use crate::{extension_indexed_docs_provider, extension_slash_command::ExtensionSlashCommand};
+
+pub struct ExtensionApi {
```

Comment:
It seems kind of weird for this type to live in the `extensions_ui` crate, as this crate is just about showing the extensions view.